backend/items/models.py

```
@@ -376,3 +409,149 @@ def donated_badge_progress(user):
+            return self.slug
+        
+        # Traverse parent chain to build path
+        path_parts = [self.slug]
+        current = self.parent
+        while current:
+            path_parts.insert(0, current.slug)
+            current = current.parent
+        
+        return '/'.join(path_parts)
+
+    def _cascade_path_update(self):
```

Ruff (0.15.2)

[warning] Missing return type annotation for private function `_cascade_path_update`

Add return type annotation: `None`

(ANN202)